sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/TextOptions.scala

```diff
@@ -39,9 +39,12 @@ private[text] class TextOptions(@transient private val parameters: CaseInsensiti
    */
   val wholeText = parameters.getOrElse(WHOLETEXT, "false").toBoolean
 
+  val lineSeparator: String = parameters.getOrElse(LINE_SEPARATOR, "\n")
+  require(lineSeparator.nonEmpty, s"'$LINE_SEPARATOR' cannot be an empty string.")
```

Comment:
This `require` looks redundant  after a `getOrElse` call

Reply:
Nope, it's `nonEmpty` on `String`.